src/client/components/editor/task-list.js

```diff
@@ -0,0 +1,164 @@
+const DirtyComponent = require('../dirty-component');
+const h = require('react-hyperscript');
+
+const { makeClassList } = require('../../../util');
+
+const Notification = require('../notification');
+const NotificationList = require('../notification/list');
+const NotificationPanel = require('../notification/panel');
+
+const getIncompleteEntities = doc => doc.entities().filter(ent => !ent.completed());
+
+
+const eleEvts = [ 'rename', 'redescribe' ];
+const entEvts = [ 'modify', 'associated', 'unassociated', 'complete', 'uncomplete'];
+const intnEvts = [ 'retype' ];
```

Comment:
You may want to simplify the events a bit.  Probably just eleEvts is fine.

You only really need 

- rename
- complete
- uncomplete

Reply:
Aside : associated is a typo : should be associate

Reply:
There is an associated event https://github.com/PathwayCommons/factoid/blob/unstable/src/model/element/entity.js#L52 and I don't remember what my line of thinking was at the time but I vaguely remember associated working better than associate.

Reply:
associate => as soon as the local action happens (optimistic)

associated => when the association is stored in the server (acknowledgement)

There may be some exceptions, but almost all UI code should only ever need the associate event.

Reply:
Okay good to know for the future.  In either case, I don't listen to those events anymore like you said.